src/components/shared/Entity/Create/index.tsx

```diff
@@ -136,9 +133,7 @@ function EntityCreate({
     ]);
 
     // TODO (defect): Trigger the prompt data loss modal if the resource config section changes.
-    // TODO (defect): Prevent prompt data loss dialog from appearing when transitioning to edit workflow.
-    const promptDataLoss = detailsFormChanged() || endpointConfigChanged();
-
+    const promptDataLoss = endpointConfigChanged();
```

Comment:
It feels like losing the name or details is not that big of a problem. I did try out just having the `detailsFormChanged` function ignoring changes to Connector... but that kind felt more wrong.